### PR TITLE
[refactor](libhdfs) introduce hadoop libhdfs

### PR DESCRIPTION
**Pull request by morningman**
# Proposed changes

Issue Number: close #xxx

## Problem summary

1. Introduce hadoop libhdfs 
2. For Linux-X86 platform, use the hadoop libhdfs
3. For other platform, use libhdfs3, because currently we don't have  hadoop libhdfs binary for other platform

Co-authored-by: adonis0147 <adonis0147@gmail.com>

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by morningman** [comment]
run buildall

**Comment by morningman** [comment]
run buildall

File: be/src/util/jni-util.cpp [hunk 1]
```diff
@@ -66,40 +66,88 @@ const std::string GetDorisJNIClasspath() {
...
-        jint res = LibJVMLoader::JNI_CreateJavaVM(&g_vm, (void**)&env, &vm_args);
+        jint res = JNI_CreateJavaVM(&g_vm, (void**)&env, &vm_args);
         if (JNI_OK != res) {
             DCHECK(false) << "Failed to create JVM, code= " << res;
         }
+
+        if (java_opts != nullptr) {
+            free(java_opts);
+        }
+        free(options);
```
Comment (adonis0147) [inline be/src/util/jni-util.cpp:150]: It's better to use RAII.

**Comment by morningman** [comment]
run buildall

**Comment by morningman** [comment]
run buildall

**Comment by morningman** [comment]
run buildall

**Review by adonis0147 — DISMISSED** [review]
LGTM

**Review by adonis0147 — APPROVED** [review]
LGTM

**Comment by morningman** [comment]
run buildall

**Pull request merged**